### PR TITLE
 test: Integration tests for workspace-types-configuration

### PR DESCRIPTION
**Pull request by ealemayehu**
Also includes test for delete-workspace , and typo fixes.

### CODE REVIEW

**Review by SanketD92 — COMMENTED** [review]
Not related to the PR. Could you also mark these tests' status in the internal spreadsheet?

**Comment by ealemayehu** [comment]
Updated the internal spreadsheet statuses of these tests.

File: main/integration-tests/__test__/api-tests/workspace-types/configurations/get-configurations.test.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    await expect(
+      researcherSession.resources.workspaceTypes
+        .workspaceType(workspaceTypeId)
+        .configurations()
+        .getAll(),
+    ).resolves.toHaveLength(0);
+  });
+
+  it('should return configurations a user has access', async () => {
```
Comment (hatimkhan) [inline main/integration-tests/__test__/api-tests/workspace-types/configurations/get-configurations.test.js]: Minor, did you mean 'should return configurations when user has access'.  Can be fixed in the next PR.

File: main/integration-tests/__test__/api-tests/workspace-types/delete-workspace-type.test.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        adminSession2.resources.workspaceTypes.workspaceType(workspaceTypeId).delete(),
+      ).rejects.toMatchObject({
+        code: errorCode.http.code.unauthorized,
+      });
+
+      await expect(adminSession.resources.workspaceTypes.workspaceType(workspaceTypeId).get()).resolves.toHaveProperty(
+        'id',
+        workspaceTypeId,
+      );
+    });
```
Comment (hatimkhan) [inline main/integration-tests/__test__/api-tests/workspace-types/delete-workspace-type.test.js:53]: Very nice, good idea to do what you just did.

File: main/integration-tests/support/resources/workspace-types/configurations/configuration.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+class Configuration extends Resource {
+  constructor({ clientSession, id, parent }) {
+    super({
+      clientSession,
+      type: 'configuration',
+      id,
+      parent,
+    });
+
+    if (_.isEmpty(parent)) throw Error('A parent resource was not provided to resource type [workspace-type]');
```
Comment (hatimkhan) [inline main/integration-tests/support/resources/workspace-types/configurations/configuration.js:29]: I think the type in the error message should say [configuration].

File: main/integration-tests/__test__/api-tests/workspace-types/configurations/get-configurations.test.js (hunk 1 above)
Comment (ealemayehu) [reply main/integration-tests/__test__/api-tests/workspace-types/configurations/get-configurations.test.js]: Fixed. Thanks for catching this!

File: main/integration-tests/__test__/api-tests/workspace-types/delete-workspace-type.test.js (hunk 2 above)
Comment (ealemayehu) [reply main/integration-tests/__test__/api-tests/workspace-types/delete-workspace-type.test.js:53]: Thanks!

File: main/integration-tests/support/resources/workspace-types/configurations/configuration.js (hunk 3 above)
Comment (ealemayehu) [reply main/integration-tests/support/resources/workspace-types/configurations/configuration.js:29]: You are correct. Fixed. Thanks!

**Review by hatimkhan — APPROVED** [review]

**Pull request merged**